### PR TITLE
Load / Store the whole object at once instead of every 8 bytes to speed things up

### PR DESCRIPTION
**Pull request by ywwu928**
# Description

## PR description

This PR _write a high-level description of your pull request_

## Checklist

- [v] The additions follow the code standards in the [developer guide](pando-rt/docs/developer.md).
- [v] New or existing tests cover these changes.
- [v] The documentation is up to date with these changes.

### CODE REVIEW

File: microbench/bfs/src/sssp.cpp [hunk 1]
```diff
@@ -12,9 +12,7 @@ void bfs::updateData(std::uint64_t val, pando::GlobalRef<std::uint64_t> ref) {
     if (val >= temp) {
       break;
     }
-  } while (!pando::atomicCompareExchange(&ref, pando::GlobalPtr<std::uint64_t>(&temp),
-                                         pando::GlobalPtr<std::uint64_t>(&val),
-                                         std::memory_order_relaxed, std::memory_order_relaxed));
+  } while (!pando::atomicCompareExchange(&ref, temp, val));
```
Comment (AdityaAtulTewari) [inline microbench/bfs/src/sssp.cpp:15]: please explain why this change is correct?

File: pando-rt/src/global_ptr.cpp [hunk 2]
```diff
@@ -226,7 +231,47 @@ void store(GlobalAddress dstGlobalAddr, std::size_t n, const void* srcNativePtr)
...
+  const auto bytesWritten = blocksWritten + numBlocks * blockSize;
+  if (remainderBytes > 0) {
+    auto byteSrc = static_cast<const std::byte*>(srcNativePtr) + bytesWritten;
+    auto byteDst = reinterpret_cast<std::uint64_t>(dstGlobalAddr) + bytesWritten;
+    for (std::size_t i = 0; i < remainderBytes; i++) {
+      auto byteData = *(byteSrc + i);
+      DrvAPI::write<std::byte>(byteDst + i, byteData);
+    }
+  }
+*/
```
Comment (AdityaAtulTewari) [inline pando-rt/src/global_ptr.cpp:274]: So why were all these lines added if they are to be commented out?

File: microbench/bfs/src/sssp.cpp (hunk 1 above)
Comment (AdityaAtulTewari) [reply microbench/bfs/src/sssp.cpp:15]: would prefer if this change was separate PR, also why is memory_order_relaxed wrong?

**Review by AdityaAtulTewari — CHANGES_REQUESTED** [review]

File: microbench/bfs/src/sssp.cpp (hunk 1 above)
Comment (AdityaAtulTewari) [reply microbench/bfs/src/sssp.cpp:15]: There is zero reason I can understand that this could work.

File: pando-rt/src/global_ptr.cpp (hunk 2 above)
Comment (ywwu928) [reply pando-rt/src/global_ptr.cpp:274]: These lines are for adding a layer of message granularity. Currently anything larger than 64 bits will crash the system, but I kept these lines for future usage is someday the issue is fixed.

File: microbench/bfs/src/sssp.cpp (hunk 1 above)
Comment (ywwu928) [reply microbench/bfs/src/sssp.cpp:15]: They are basically the exact same code. Just like Max said in Discord, usual atomic operations should only contain one pointer or address and the rest operand should be scalar numbers. There were already multiple versions of atomicCompareExchange, one takes GlobalPtr as operands and the other takes scalar as operands. I figured out there is no reason for me to intentionally transfom from scalar to GlobalPtr in this case here.

Comment (ywwu928) [reply microbench/bfs/src/sssp.cpp:15]: There is nothing wrong with memory_ordered_relaxed. It's just that the scalar operand version does no take memory order as argument. The default order is __ATOMIC_SEQ_CST (total ordering) by the function.

File: pando-rt/src/global_ptr.cpp (hunk 2 above)
Comment (AdityaAtulTewari) [reply pando-rt/src/global_ptr.cpp:274]: ah okay add that to the comment please

Comment (AdityaAtulTewari) [reply pando-rt/src/global_ptr.cpp:274]: Add the comment and then merge main into this so I can approve this pr.

**Review by AdityaAtulTewari — APPROVED** [review]
